backend/routes/api/components/componentUtils.js

```diff
@@ -152,11 +153,14 @@ const getEnabledConfigMaps = (fastify, appDefs) => {
 const getApplicationDefs = () => {
   const normalizedPath = path.join(__dirname, '../../../../data/applications');
   const applicationDefs = [];
+  const featureFlags = features.getComponentFeatureFlags();
   fs.readdirSync(normalizedPath).forEach((file) => {
     if (constants.yamlRegExp.test(file)) {
       try {
         const doc = jsYaml.load(fs.readFileSync(path.join(normalizedPath, file), 'utf8'));
-        applicationDefs.push(doc);
+        if (!doc.spec.featureFlag || featureFlags[doc.spec.featureFlag]) {
```

Comment:
future: this may also work with just the one check and optional chaining, I'd need to fully confirm. Regardless, what you have works, pre-opt chaining